test/models/user_test.rb

```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  def setup
+    @github_omniauth_hash = OmniAuth.config.mock_auth[:github]
+    @user                 = users(:jim)
+  end
+
+  context 'validations' do
+    context 'presence' do
+      [:provider, :uid, :login, :email, :token].each do |column|
+        should validate_presence_of(column)
+      end
+    end
+
+    context 'uniqueness' do
+      [:uid, :login, :email, :token].each do |column|
+        should validate_uniqueness_of(column)
+      end
+    end
+  end
+
+  test 'create a new valid user from the auth hash' do
```

Comment:
If I use `context` then the test looks like this. This is from the [shoulda gem](https://github.com/thoughtbot/shoulda)

```ruby
context '#create_from_auth_hash' do
  should 'create valid user' do
    assert_difference 'User.count', 1 do
      User.create_from_auth_hash(@github_omniauth_hash)
    end
  end
end
```

Now that I look at it again, I agree `should` doesn't  look quite right as part of a test.

This is what I'm going to use for my tests, this reads better to me (I'm not a big fan of nesting).
```ruby
test '#create_from_auth_hash creates a valid user' do
  assert_difference 'User.count', 1 do
    User.create_from_auth_hash(@github_omniauth_hash)
  end
end
```